lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -1229,6 +1232,7 @@ private TaskImpl terminateCancelCommonActions(String taskId, TaskState targetSta
     task.setModified(now);
     task.setCompleted(now);
     task.setState(targetState);
+    task = (TaskImpl) taskEndstatePreprocessorManager.processTaskBeforeEndstate(task);
```

Comment:
Please check if the SPI is enabled and only if that's the case execute this line of code

Reply:
We do this check inconsistenly for other spis. For example, the line ` task = (TaskImpl) beforeRequestReviewManager.beforeRequestReview(task); ` is executed unconditionally. Same as for TaskEndstatePreprocessor, calling the SPIManager returns the same Task if there is no SPI. That's why it's not necessary to check if the SPI is enabled.  However, some other SPIs are executed conditionally. I think it would be better to make it consistently. Should I change all SPIs that are not checked? Or what would be the best solution?

Reply:
I took a closer look at this and it's true. It seems to me, that some SPIs really need the isEnabled-Check and some don't absolutely have to. The ones that don't have to are the ones modifying the task and returning it. The ones that have to are the ones that perform logic additionally to the actual SPI call (performance) e.g. history

However, if we decide to use it the way it is implemented here, do we need the isEnabled method for this SPI at all? The API you mentioned (beforeRequestReview(task)) doesn't have it. All others do and use it either in the code or in the manager itself.

Maybe it's easier to have a short call about this together